chia/wallet/trade_manager.py

```diff
@@ -254,10 +256,11 @@ async def cancel_pending_offer_safely(
                 all_txs.append(tx)
             else:
                 # ATTENTION: new_wallets
-                txs = await wallet.generate_signed_transaction(
-                    [coin.amount], [new_ph], fee=fee_to_pay, coins={coin}, ignore_max_send_amount=True
-                )
-                all_txs.extend(txs)
+                if isinstance(wallet, (CATWallet, NFTWallet, DataLayerWallet)):
+                    txs = await wallet.generate_signed_transaction(
+                        [uint64(coin.amount)], [new_ph], fee=fee_to_pay, coins={coin}, ignore_max_send_amount=True
+                    )
+                    all_txs.extend(txs)
```

Comment:
Should this be an `elif ...:`?  Maybe with an `else:` still to explicitly catch not-yet-considered wallets?  Seems like an exception causing failure would be better than a comment.

Reply:
Yea, I'll need to study this a little more because I'm not entirely sure what would/should happen with not-yet-considered wallets. It should probably raise somewhere , I'm just not sure where